pkg/controller/job/job_controller.go

```diff
@@ -986,13 +996,21 @@ func ensureJobConditionStatus(list []batch.JobCondition, cType batch.JobConditio
 				list[i].Reason = reason
```

Comment:
So I did begin to change LastProbeTime, but there's a problem with that. By definition, LastProbeTime must be updated every time ensureJobConditionStatus is called regardless of whether the status/reason/message matches (because it's just a probe). That means that jobConditionsChanged is always true every time syncJob is called. That means that we'll do a `Jobs.UpdatedStatus()` every time syncJob is called and that `forget` is always true. These are significantly more far-reaching consequences than I had originally anticipated.

The performance impact of syncing the Job with the API server every time syncJob is called could be disastrous for long running jobs that see very infrequent updates.

So I'd like to hold off from making LastProbeTime change in this PR. It also sends the wrong message to update it only when job condition status changes. AFAICT there are also no other k8s controllers that do anything like LastProbeTime, so I think removing that field might even be the best course of action.

Reply:
I was thinking that if the block in line 643 is reached, you can update all `LastProbeTime`s